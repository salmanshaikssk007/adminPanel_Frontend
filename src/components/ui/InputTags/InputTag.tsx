import React from 'react';
import Styles from './InputTag.module.css';

interface props {
  type: React.HTMLInputTypeAttribute;
  placeholder?: string;
  inputStylesName?: string;
  handleEvent?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  value?: string | number | readonly string[] | undefined;
  disabled?: boolean;
}

const InputTag = (props: props) => {
  //check if react ref requried
  const variousName = props.inputStylesName;
  return (
    <div className={Styles.inputTag}>
      <input
        disabled={props.disabled}
        value={props.value}
        onChange={props.handleEvent}
        data-testid="input_ui"
        className={`${Styles.variousName}`}
        type={props.type}
        placeholder={props.placeholder}
      />
    </div>
  );
};

export default InputTag;
