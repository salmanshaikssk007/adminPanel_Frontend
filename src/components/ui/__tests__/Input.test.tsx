import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import InputTag from '../InputTags/InputTag';

configure({ adapter: new Adapter() });

describe('<InputTag />', () => {
  test('check of type', () => {
    const wrapper = shallow(<InputTag type={'text'} />);

    const input = wrapper.find('input');
    expect(input.type()).toBe('input');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
