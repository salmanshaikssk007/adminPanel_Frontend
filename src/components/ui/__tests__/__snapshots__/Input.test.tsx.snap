// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputTag /> check of type 1`] = `"<div class=\\"inputTag\\"><input data-testid=\\"input_ui\\" class=\\"variousName\\" type=\\"text\\"/></div>"`;
